x-pack/plugins/security/server/fips/fips_service.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger } from '@kbn/logging';
+import type { SecurityLicense } from '@kbn/security-plugin-types-common';
+
+import type { ConfigType } from '../config';
+
+export interface FipsServiceSetupParams {
+  config: ConfigType;
+  license: SecurityLicense;
+}
+
+export interface FipsServiceSetupInternal {
+  validateLicenseForFips: () => void;
+}
+
+export class FipsService {
+  private readonly logger: Logger;
+  private isInitialLicenseLoaded: boolean;
+
+  constructor(logger: Logger) {
+    this.logger = logger;
+    this.isInitialLicenseLoaded = false;
+  }
+
+  setup({ config, license }: FipsServiceSetupParams): FipsServiceSetupInternal {
+    return {
+      validateLicenseForFips: () => this.validateLicenseForFips(config, license),
+    };
+  }
+
+  private validateLicenseForFips(config: ConfigType, license: SecurityLicense) {
+    license.features$.subscribe({
+      next: (features) => {
+        const errorMessage = `Your current license level is ${license.getLicenseType()} and does not support running in FIPS mode.`;
+
+        if (license.isLicenseAvailable() && !this.isInitialLicenseLoaded) {
+          if (config?.fipsMode.enabled && !license.getFeatures().allowFips) {
+            this.logger.error(errorMessage);
+            throw new Error(errorMessage);
```

Comment:
Logging the error is redundant since an Error is being thrown, but I could not find a better way to be able to assert this in the unit tests since `throw Error` from inside a SafeSubscriber doesn't work the same as throwing an error from the function itself.

Reply:
I tried a few different things here with no luck. Maybe there is a way to use a jest spy? I found this, but didn't have time to play with it: https://stackoverflow.com/questions/69128106/how-to-unit-test-code-inside-subscribe-using-jest